std/node/url.ts

```diff
@@ -0,0 +1,1197 @@
+import {
```

Comment:
I imagine Node's URL is pretty similar to Deno's built-in global URL object. Can you not use that? Ideally we wouldn't have to maintain an independent fork of this module.

Reply:
Do you mean [this module](https://github.com/denoland/deno/blob/master/cli/js/web/url.ts) ?
If so, they're not exactly the same, the global URL uses the WHATWG standard, but the node's url does not
And node's url has a different functions, like : `format , fileURLToPath , resolveObject , ...` in general , which i mentioned before

Reply:
But you could use the built-in URL class to implement some parts of the Node module?

Reply:
I worked on it, the result was not very good and clean in my eyes , 
Because the behavior of the general version is structurally different from the node's version

Finally, the client will feel the difference in the code in deno and node

Reasons :
1. built-in URL doesn't work with empty or "/" url for base input but node's version will
```javascript
//node js
const url = require("url");
new url.parse("/"); // it's ok and return a Url object
```

```javascript
//deno's and node general URL module
new URL("/"); // it's throw an error -> TypeError: URL requires a base URL.
```

2. node's url parser has 2 more arguments `[ parseQueryString and slashesDenoteHost ]` than general one ,  That affects `parsing operations`, and if I want to rewrite their only common function ( parsing ), I've only made the code uglier.

3. the other parts are not in common and can't implement from general URL

For this reasons , the node behaves in the same way and has two URL versions, one exactly like the [deno general version](https://github.com/denoland/deno/blob/master/cli/js/web/url.ts),

and another one like this one on this PR , totally separate from each other

Reply:
If you don't agree with the node's old version
I will delete it and And I leave only independent functions `[ fileURLToPath ,  pathToFileURL ]`